pkg/util/strings.go

```diff
@@ -29,3 +34,93 @@ func StringsClone(s string) string {
 	copy(b, s)
 	return *(*string)(unsafe.Pointer(&b))
 }
+
+// MergeSlicesParallel merge sorted slices in parallel
+// using the MergeSlicesV2 function
+func MergeSlicesParallel(parallelism int, a ...[]string) []string {
+	if parallelism <= 1 {
+		return MergeSlicesV2(a...)
+	}
+	if len(a) == 0 {
+		return nil
+	}
+	if len(a) == 1 {
+		return a[0]
+	}
+	c := make(chan []string, len(a))
+	wg := sync.WaitGroup{}
+	var r [][]string
+	p := min(parallelism, len(a)/2)
+	batchSize := len(a) / p
+
+	for i := 0; i < len(a); i += batchSize {
+		wg.Add(1)
+		go func(i int) {
+			m := min(len(a), i+batchSize)
+			c <- MergeSlicesV2(a[i:m]...)
+			wg.Done()
+		}(i)
+	}
+
+	go func() {
+		wg.Wait()
+		close(c)
+	}()
+
+	for s := range c {
+		r = append(r, s)
+	}
+
+	return MergeSlicesV2(r...)
+}
+
+func NewStringListIter(s []string) *StringListIter {
+	return &StringListIter{l: s}
+}
+
+type StringListIter struct {
+	l   []string
+	cur string
+}
+
+func (s *StringListIter) Next() bool {
+	if len(s.l) == 0 {
+		return false
+	}
+	s.cur = s.l[0]
+	s.l = s.l[1:]
+	return true
+}
+
+func (s *StringListIter) At() string { return s.cur }
+
+var MAX_STRING = string([]byte{0xff})
+
+// MergeSlicesV2 merges a set of sorted string slices into a single ones
+// while removing all duplicates.
+func MergeSlicesV2(a ...[]string) []string {
```

Comment:
Why it is named V2? Can we have a better name for it

Reply:
I can rename! :D Just did this as a way to differentiate the one from thanos and forget about it.

Reply:
done